packages/js-sdk/src/message/renderer/messageTypes/renderDelete.ts

```diff
@@ -0,0 +1,62 @@
+import { MessageType } from '@dm3-org/dm3-lib-messaging';
+import { MessageActionType, MessageModel } from '../../types';
+
+export const renderDelete = (messages: MessageModel[]) => {
+    //We filter out all messages that are deleted
+    const deleteMessages = messages.filter(
+        (message) => message.envelop.message.metadata.type === 'DELETE_REQUEST',
+    );
+
+    //We get the message hash of the messages that are to be deleted
+    const toBeDeletedByMessageHash = deleteMessages.map(
+        (deleteMessage) =>
+            deleteMessage.envelop.message.metadata.referenceMessageHash,
+    );
+
+    //We return the messages with out messages that are to be deleted and the delete requests
+    return messages
+        .filter(
+            (message) =>
+                message.envelop.message.metadata.type !==
+                MessageActionType.DELETE,
+        )
+        .map((message) => {
+            if (
+                toBeDeletedByMessageHash.includes(
+                    message.envelop.metadata?.messageHash,
+                )
+            ) {
+                return {
+                    ...message,
```

Comment:
looks like we might wrap this in a some kind of merge function or even use lodash merge function here.

Reply:
what do you mean by merge function? Are you referring to the fact that properties nested into the object are subject to change?